src/hw/installApp.js

```diff
@@ -20,9 +20,14 @@ export default function installApp(
     firmwareKey: app.firmware_key,
     hash: app.hash,
   }).pipe(
-    filter((e) => e.type === "bulk-progress"), // only bulk progress interests the UI
+    filter(
+      (e) =>
+        e.type === "bulk-progress" || // only bulk progress interests the UI
+        e.type === "device-permission-requested" || // exposing these for inline install
+        e.type === "device-permission-granted"
+    ),
     throttleTime(100), // throttle to only emit 10 event/s max, to not spam the UI
-    map((e) => ({ progress: e.progress })), // extract a stream of progress percentage
+    map((e) => (e.progress ? { progress: e.progress } : e)),
```

Comment:
i'm a bit scared by consequence of this & actually i feel we should maybe no longer remap the event to a `{progress}` specifically because that `e` of type "bulk-progress" already have a `.progress` (so tbd but shouldn't break that much). however, if you don't do it, then you have no clean way to "switch on type".